### DISCUSSION TITLE
DLL size

### DISCUSSION THREAD

**Discussion by yonist** [duckdb/duckdb#6818] [General]
First of all thank you for the amazing work... 

I noticed that the DLL/SO files are big. While for sure I cannot compare 2 different application because of different functionality just to give you a sense of the issue sqlite.dll is about 3mb and duckdb is about 25mb on windows (41mb on linux)

I think something is not configured right. For example in the attached image I look at the binary and you can see there is html/css/javascript embedded directly into the DLL 

![image](https://user-images.githubusercontent.com/499992/226857672-0d34b125-4f59-4e22-9372-69bbc58ec81f.png)

**Comment by yonist** [comment]
There is also this text in the dll

![image](https://user-images.githubusercontent.com/499992/226864193-9ffce141-29c7-46ee-97b2-9d1260c4c83f.png)

**Reply by Tishj** [reply]
That looks like it comes from `duckdb/extension/tpcds/dsdgen/answers/sf1/98.csv`

**Comment by Mytherin** [comment]
It is expected that DuckDB is bigger than SQLite since DuckDB is much more "batteries included" than SQLite. DuckDB implements many more functions, data types, SQL functionality, etc. than SQLite. This is especially true for pre-packaged variants that include extensions such as the Parquet reader, which then include libraries such as thrift, snappy, zstd, etc. DuckDB also contains templated loops for optimized vectorized execution which speed up execution at the expense of binary size.

Nevertheless we can push down the size somewhat and we have some ideas on how to do that. It might also be worthwhile to remove certain extensions from our distributions (e.g. the Python client probably should not ship with the tpch, tpcds and visualizer extensions - those extensions is where the text you are mentioning comes from). I would not expect DuckDB to ever be as small as SQLite, however, since that would require us to strip down the functionality provided by DuckDB immensely.

**Reply by yonist** [reply]
Thank you for considering it.
